plugins/network-elements/nuage-vsp/src/com/cloud/network/guru/NuageVspGuestNetworkGuru.java

```diff
@@ -129,66 +134,94 @@ public Network design(NetworkOffering offering, DeploymentPlan plan, Network use
-            vpcName = vpcObj.getName();
-            vpcUuid = vpcObj.getUuid();
-        }
+            if (isVpc) {
+                Vpc vpcObj = _vpcDao.findById(vpcId);
+                vpcName = vpcObj.getName();
+                vpcUuid = vpcObj.getUuid();
+            }
+
+            if (isSharedNetwork) {
+                List<VlanVO> vlans = _vlanDao.listVlansByNetworkId(network.getId());
+                for (VlanVO vlan : vlans) {
+                    boolean isIpv4 = StringUtils.isNotBlank(vlan.getIpRange());
+                    String[] range = isIpv4 ? vlan.getIpRange().split("-") : vlan.getIp6Range().split("-");
+                    ipAddressRanges.add(range);
+                }
+            }
 
-        HostVO nuageVspHost = getNuageVspHost(physicalNetworkId);
-        ImplementNetworkVspCommand cmd = new ImplementNetworkVspCommand(networksDomain.getName(), networksDomain.getPath(), networksDomain.getUuid(),
-                networksAccount.getAccountName(), networksAccount.getUuid(), network.getName(), network.getCidr(), network.getGateway(), network.getUuid(), isL3Network, vpcName,
-                vpcUuid, offering.getEgressDefaultPolicy(), ipAddressRange);
-        ImplementNetworkVspAnswer answer = (ImplementNetworkVspAnswer)_agentMgr.easySend(nuageVspHost.getId(), cmd);
+            HostVO nuageVspHost = getNuageVspHost(physicalNetworkId);
+            List<String> dnsServers = _nuageVspManager.getDnsDetails(network);
+            List<String> gatewaySystemIds = _nuageVspManager.getGatewaySystemIds();
+            ImplementNetworkVspCommand.Builder cmdBuilder = new ImplementNetworkVspCommand.Builder().networkDomainName(networksDomain.getName()).networkDomainPath(networksDomain.getPath())
+                    .networkDomainUuid(networksDomain.getUuid()).networkAccountName(networksAccount.getAccountName()).networkAccountUuid(networksAccount.getUuid())
+                    .networkName(network.getName()).networkCidr(network.getCidr()).networkGateway(network.getGateway()).networkAclId(network.getNetworkACLId()).dnsServers(dnsServers)
+                    .gatewaySystemIds(gatewaySystemIds).networkUuid(network.getUuid()).isL3Network(isL3Network).isVpc(isVpc).isSharedNetwork(isSharedNetwork).vpcName(vpcName)
+                    .vpcUuid(vpcUuid).defaultEgressPolicy(offering.getEgressDefaultPolicy()).ipAddressRange(ipAddressRanges).domainTemplateName(preConfiguredDomainTemplateName);
+            ImplementNetworkVspAnswer answer = (ImplementNetworkVspAnswer)_agentMgr.easySend(nuageVspHost.getId(), cmdBuilder.build());
 
-        if (answer == null || !answer.getResult()) {
-            s_logger.error("ImplementNetworkNuageVspCommand failed");
-            if ((null != answer) && (null != answer.getDetails())) {
-                s_logger.error(answer.getDetails());
+            if (answer == null || !answer.getResult()) {
+                s_logger.error("ImplementNetworkNuageVspCommand failed");
```

Comment:
Please provide context information about the Nuage device and network that failed to configure.